### PR TITLE
Passing term to the server request termdb/categories

### PR DESCRIPTION
**Pull request by airenzp**
## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/termdb/TermTypeSearch.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ The tree target is used to determine the allowed term types.
 
 const useCases = {
 	matrix: [TermTypeGroups.DICTIONARY_VARIABLES, TermTypeGroups.MUTATION_CNV_FUSION],
-	filter: [TermTypeGroups.DICTIONARY_VARIABLES, TermTypeGroups.MUTATION_CNV_FUSION],
+	filter: [TermTypeGroups.DICTIONARY_VARIABLES, TermTypeGroups.MUTATION_CNV_FUSION, TermTypeGroups.GENE_EXPRESSION],
```
Comment (xzhou82) [inline client/termdb/TermTypeSearch.ts:17]: please remove geneExp from filter, and do it in next branch where client/filter/tvs.geneExpression.js will be created. right now breaks

Comment (airenzp) [reply client/termdb/TermTypeSearch.ts]: I just did. It was pushed by mistake. Please delete the branch and pull again

File: client/filter/tvs.js [hunk 2]
```diff
@@ -59,7 +60,7 @@ class TVS {
 	async setHandler() {
 		if (!this.tvs || !this.tvs.term) return
 		const term = this.tvs.term
-		const type = term.type == 'integer' || term.type == 'float' ? 'numeric' : term.type // 'categorical', 'condition', 'survival', etc
+		const type = isNumericTerm(term.type) ? 'numeric' : term.type // 'categorical', 'condition', 'survival', etc
```
Comment (xzhou82) [inline client/filter/tvs.js:63]: typo. do `isNumericTerm(term)`

Comment (airenzp) [reply client/filter/tvs.js]: done

File: server/routes/termdb.categories.ts [hunk 3]
```diff
@@ -89,43 +83,19 @@ function init({ genomes }) {
...
-			term.stop = q.stop
-		}
-	} else {
-		term = tdb.q.termjsonByOneid(q.tid)
-	}
-	const tw =
-		q.type == 'geneVariant'
-			? { $id: q.gene || q.name || q.tid, term, q: { isAtomic: true } }
-			: { $id: q.tid, id: q.tid, term, q: q.term1_q || getDefaultQ(term, q) }
+	const tw: TermWrapper = { term: q.term, q: q.term1_q || getDefaultQ(q.term, q) }
```
Comment (xzhou82) [inline server/routes/termdb.categories.ts:98]: how about we still create $id with random string so later when deleting auto-assigning tw.$id from getData(), this code won't need change

File: server/routes/termdb.categories.ts [hunk 4]
```diff
@@ -134,10 +104,12 @@ async function trigger_getcategories(
 	}
 
 	const data = await getData(arg, ds, genome)
+	const id = tw.$id!
```
Comment (xzhou82) [inline server/routes/termdb.categories.ts:107]: can delete this

File: server/routes/termdb.categories.ts [hunk 5]
```diff
@@ -149,7 +121,7 @@ async function trigger_getcategories(
 		}
 		const sampleCountedFor = new Set() // if the sample is counted
 		for (const [sampleId, sampleData] of Object.entries(samples)) {
-			const key = tw.$id //getData indexes terms by tw.$id, that may be $id or the term id or the name
+			const key = tw.$id! //getData inits tw.$id to the term id or the name as it is empty
```
Comment (xzhou82) [inline server/routes/termdb.categories.ts:124]: can delete this line without creating new variable `key`, and just use tw.$id in subsequent lines

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**